### ISSUE TITLE
Structural editing toggle broke all my code

### ISSUE THREAD

**Issue by mikub** [cursive-ide/cursive#1669]
Gave Cursive a chance and was quite enthusiastic fan but now I am very very scared. Toggling Structural editing (especially to Parinfer Indent Mode) automatically (!) made changes in arbitrary places of my entire codebase (!!!) without any prompt (!!!!). 
It might be some rookie mistake (though i am quite experienced with other java IDEs - eclipse/STS - as well as with clojure IDEs like LightTable). Seems that even files that i did not edit/save (!) were affected. Closing brackets/parens moved all over the places in many many of my functions, especially in multiple-arity functions.

I am sorry but i will not even try to reproduce this or write a test case (though I am able to)  as I am going to run very very far and very very fast away from Cursive. It is the first time in my life were my IDE made unwanted changes to my entire project, changing code arbitrarily in various places with no prompt! Going to go into my svn to grab last commit and then off back to emacs or light table. It's a shame as i often work on java projects or interoperability and i miss the full blown IDEs features. I guess I will not be seeing you Cursive anytime soon, so live long and prosper.

**Comment by mikub** [comment]
Here is a diff with base to illustrate changes that were made by Cursive 
![cursive bug](https://cloud.githubusercontent.com/assets/674012/23339251/2bca0288-fc59-11e6-8792-598709b54d66.png)

**Comment by cursive-ide** [comment]
Several people have reported this issue, but I've never been able to reproduce it. Thanks for the diff, I'll see if I can figure out what's going on based on that. If you have more diffs they would be appreciated.

**Comment by mikub** [comment]
Here you go :)

![cursive bug2](https://cloud.githubusercontent.com/assets/674012/23350111/2e27db0a-
fcf3-11e6-8978-4438c24bd3e2.png)
![cursive bug3](https://cloud.githubusercontent.com/assets/674012/23350138/64ccbae0-fcf3-11e6-9228-cdcaeac929ab.png)

**Comment by mikub** [comment]
![cursive bug2](https://cloud.githubusercontent.com/assets/674012/23350162/98b27dcc-fcf3-11e6-8084-5d3163282cdc.png)

**Comment by sveri** [comment]
I just had a similar bug with cursive, parinfer and a multi arity function, which should be easy to reproduce, hopefully.

This is my function working:

    (defn get-from-group-for-field [field entites db db-data]
      [:div.form-group
       [:label (u/prepare-str-for-ui (:name field))]
       (get-form-for-type field entites db db-data)])))

And when I convert it to a multi arity function:

    (defn get-from-group-for-field
      ([field entites db] (get-from-group-for-field field entites db nil))
      ([field entites db db-data
        [:div.form-group
         [:label (u/prepare-str-for-ui (:name field))]
         (get-form-for-type field entites db db-data)]]))))

You can see that the parameters of the second function always moves the _]_ at the end of the function, which is incorrect. 

Maybe you can reproduce that :-)

**Comment by codeboost** [comment]
Steps to reproduce:
1. Disable parinfer
2. Edit a clojure file, change the indentation, move a parenthesis to next line, etc. 
3. Save and close the file.
4. Enable parinfer
5. Open the file (or any other badly formatted file), then switch to another file, wait a couple of seconds.
6. Close the opened file.

Notice that the file has been modified and saved with no prompt. 

So opening a badly formatted file causes `parinfer` to reformat it automatically and I guess IntelliJ auto saves it.
 
This is really annoying when working on code bases with people who don't use `parinfer`.
Just opening files causes reformatting, sometimes changing the semantics of the code.

Possible solutions:
Per-file structural editing as an option: Default off, enabled manually for each file.  

Parinfer disabled until the user actually modifies the file, so no structural editing when viewing files.

**Comment by cursive-ide** [comment]
The fact that parinfer modifies files when opening them is intrinsic to parinfer, at least up till now. I actually just wrote to the mailing list this morning discussing all this [here](https://groups.google.com/d/msg/cursive/JpjS9eYIBAc/BqPmnDyZCQAJ). The tl;dr is that this should hopefully be fixed soon with the integration of parinfer v3, which comes (in Cursive) with a new way of integrating parinfer I've developed. There's also a link there to a doc I wrote in the parinfer wiki describing why the occasional breakage (the original problem of this ticket) happens.
